### ISSUE TITLE
Python error on area loading screens 

### ISSUE THREAD

**Issue by MarcelHB** [gemrb/gemrb#1521]
For at least PST and IWD, a Python error appears when changing areas:

```
[GUIScript/ERROR]: Runtime Error:
[Python/ERROR]: AttributeError: 'GWindow' object has no attribute 'SCRIPT_GROUP'
[Python/ERROR]: RuntimeError: Invalid Scripting reference, must have SCRIPT_GROUP attribute.
[Python/ERROR]:
The above exception was the direct cause of the following exception:
[Python/ERROR]: Traceback (most recent call last):
[Python/ERROR]:   File "D:/Sources\gemrb\gemrb\GUIScripts\pst/LoadScreen.py", line 52, in EndLoadScreen
[Python/ERROR]:     Skull = Window.GetControl (1)
[Python/ERROR]:   File "D:/Sources\gemrb\gemrb\GUIScripts/GUIClasses.py", line 211, in GetControl
[Python/ERROR]:     return GetView(self, newID)
[Python/ERROR]: SystemError: <built-in function GetView> returned a result with an error set
```
I can't find anything breaking so far, yet it better goes away.

**Comment by lynxlynxlynx** [comment]
I can't reproduce this any more anywhere, but in iwd2 a different error is visible on a reload, which I'll address:
```
[GUIScript/ERROR]: Runtime Error:                                                                                 
[Python/ERROR]: Traceback (most recent call last):                                                                
[Python/ERROR]:   File "../gemrb/GUIScripts/MetaClasses.py", line 58, in handler                                  
[Python/ERROR]:     return f(*args, **kwargs)                                                                     
[Python/ERROR]: RuntimeError: Can't find GameControl!                                                             
[Python/ERROR]: 
During handling of the above exception, another exception occurred:                                               
[Python/ERROR]: Traceback (most recent call last):                                                                
[Python/ERROR]:   File "../gemrb/GUIScripts/GUICommonWindows.py", line 1373, in <lambda>                          
[Python/ERROR]:     window.SetAction(lambda: TopWindowClosed(window), ACTION_WINDOW_CLOSED)                       
[Python/ERROR]:   File "../gemrb/GUIScripts/GUICommonWindows.py", line 1347, in TopWindowClosed                   
[Python/ERROR]:     SelectionChanged()                                                                            
[Python/ERROR]:   File "../gemrb/GUIScripts/GUICommonWindows.py", line 1914, in SelectionChanged                  
[Python/ERROR]:     UpdateActionsWindow ()                                                                        
[Python/ERROR]:   File "../gemrb/GUIScripts/GUICommonWindows.py", line 641, in UpdateActionsWindow                
[Python/ERROR]:     CurrentWindow.SetupControls (globals(), pc, ActionBarControlOffset)                           
[Python/ERROR]:   File "../gemrb/GUIScripts/MetaClasses.py", line 60, in handler                                  
[Python/ERROR]:     raise type(e)(str(e) + "\nMethod Docs:\n" + str(f.__doc__))                                   
[Python/ERROR]: RuntimeError: Can't find GameControl!  
```

**Comment by MarcelHB** [comment]
I still see this. But take your turn first then, I'll have another look after that.

**Comment by lynxlynxlynx** [comment]
I see now I didn't read your report  carefully and only tested reloading, not normal area movement.

**Comment by lynxlynxlynx** [comment]
Using GetView to refetch the window by alias returns None, so I guess it gets invalidated or killed by core earlier (perhaps DestroyAllWindows). Just returning early works, though I can't say I even see a loading screen on area moves, which could be from reaching 100% so fast. The code definitely runs, since iwd2 is also appending hint text.

**Comment by bradallred** [comment]
when you load the game, `EndLoadScreen` is what closes the window (`GemRB.SetTimer(LoadScreen.Close, 500, 0)`). I dont know if/why that would change for changing areas, but maybe we don't care. As you say, the loads are so fast there isn't a point.

**Comment by lynxlynxlynx** [comment]
Yeah, but that closing is after the error line, that's the odd bit. If you comment it out on area move, no window is visible ... since StartLoadScreen never ran in the first place. We just call the closing function. Ah, at least we're getting somewhere — will take a look if we should be triggering Game::LoadMap differently, if showing the screen makes sense at all. If not, it's still clear how to fix this.

**Comment by lynxlynxlynx** [comment]
So we're currently deliberately not showing loading screens after you're in the game. If you count the calls, it makes sense at the start:
StartLoadScreen, StartLoadScreen, EndLoadScreen

Which is fine, since we change the picture and quote midprogress, I assume like the originals. But then on area move, just: 
EndLoadScreen

Which makes sense in the part that we didn't start a load screen. But not who called the function, how is the old progress bar still alive?  I wouldn't expect us to enter this function.

I tried making the LOAD_PROG aliasing overwriting, but that didn't change anything. Tried RemoveScriptingRef on window close, realiasing, reseting the event handler, but none had any effect. RemoveScriptingRef needs a view, not an alias, so I imagine the view is gone by then. I see the View destructor clears all of them, but I guess the window stays around, since we just close it, not destroy it?

Trying to remove just that one alias or even view in core has no effect (removing the control would be wrong any way):
```
diff --git a/gemrb/core/Interface.cpp b/gemrb/core/Interface.cpp
index 5582f44f6..633d6081d 100644
--- a/gemrb/core/Interface.cpp
+++ b/gemrb/core/Interface.cpp
@@ -3405,6 +3405,8 @@ void Interface::LoadProgress(int percent)
        if (prog) {
                // loadwin is NULL when LoadMap is called and passes false for the loadscreen param
                prog->SetValue(percent);
+               Window* loadWin = GetControl<Window>("LOADWIN", 0);
+               if (percent == 100 && loadWin) loadWin->RemoveScriptingRef(prog->GetScriptingRef()); // loadWin->RemoveSubview(prog);
        }
 
        video->SwapBuffers();
```

So I'm stumped at what's going on. Working around the problem is easy, but I wanted a clean fix.

**Comment by bradallred** [comment]
> but I guess the window stays around, since we just close it, not destroy it?

because in general you can't destroy a window upon close. there is more than one reason for it, but consider that `OnClose` callbacks couldn't work as the main one. It's been this way since long before subviews too. there is some relevant commentary above `WindowManager::CloseWindow`. Closing a window _does_  clear the `Window` scripting refs (assuming the `DestroyOnClose` flag is set), but that won't clear the refs of the subviews, nor do we want it to do so since `OnClose` callbacks often want to use them.

TL;DR: Windows only get destroyed when a new window is opened, so if you are entering a game and then immediately going to a new area that is why the loading window still exists.

**Comment by lynxlynxlynx** [comment]
Sure, but how to disable the callback in this case? It seems making the control disabled works, too bad I didn't try it earlier.

**Comment by bradallred** [comment]
seems like the `LoadScreen`  `OnClose` callback needs to clear the bar alias so that `Interface::LoadProgress` doesnt try to set its value. `GemRB.GetView("LOAD_PROG").RemoveScriptingRef()` should work.

**Comment by lynxlynxlynx** [comment]
I tried that. Anyway, the fix in core is a oneliner, so it's fine.
